// @storybook/svelte/dist/cjs/client/preview/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.storiesOf = exports.setAddon = exports.raw = exports.getStorybook = exports.forceReRender = exports.configure = exports.clearDecorators = exports.addParameters = exports.addDecorator = void 0;

var _core = require("@storybook/core");

var _decorators = require("./decorators");

require("./globals");

var _render = require("./render");

var _start = (0, _core.start)(_render.renderToDOM, {
  decorateStory: _decorators.decorateStory,
  render: _render.render
}),
    coreConfigure = _start.configure,
    clientApi = _start.clientApi,
    forceReRender = _start.forceReRender;

exports.forceReRender = forceReRender;
var setAddon = clientApi.setAddon,
    addDecorator = clientApi.addDecorator,
    addParameters = clientApi.addParameters,
    clearDecorators = clientApi.clearDecorators,
    getStorybook = clientApi.getStorybook,
    raw = clientApi.raw;
exports.raw = raw;
exports.getStorybook = getStorybook;
exports.clearDecorators = clearDecorators;
exports.addParameters = addParameters;
exports.addDecorator = addDecorator;
exports.setAddon = setAddon;
var framework = 'svelte';

var storiesOf = function storiesOf(kind, m) {
  return clientApi.storiesOf(kind, m).addParameters({
    framework: framework
  });
};

exports.storiesOf = storiesOf;

var configure = function configure(loadable, m) {
  return coreConfigure(framework, loadable, m);
};

exports.configure = configure;
